### PR TITLE
Feature/solidity 0.6.1

### PR DESCRIPTION
**Pull request by krlosMata**
- Upgrade contracts to solidity 0.6.1
  - breaking changes solidity 0.6.X: https://solidity.readthedocs.io/en/v0.6.0/060-breaking-changes.html
- Add `solidity-coverage` tool
  - https://github.com/sc-forks/solidity-coverage

### CODE REVIEW

File: contracts/Rollup.sol [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 import './RollupInterface.sol';
 import './VerifierInterface.sol';
 
 /**
  * @dev Define interface ERC20 contract
  */
-contract ERC20 {
-    function transfer(address recipient, uint256 amount) external returns (bool);
-    function transferFrom(address sender, address recipient, uint256 amount) external returns (bool);
+abstract contract ERC20 {
```
Comment (invocamanman) [inline contracts/Rollup.sol]: I think it should be `interface ERC20` like in https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/IERC20.sol

File: contracts/Rollup.sol [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+    function transfer(address recipient, uint256 amount) external virtual returns (bool);
```
Comment (invocamanman) [inline contracts/Rollup.sol]: as it says in docs: "Add virtual to every non-interface function you intend to override" so if we defined as an interface we shouldn't use the `virtual` key word

File: contracts/Rollup.sol (hunk 1 above)
Comment (krlosMata) [reply contracts/Rollup.sol]: change it ! xD

File: contracts/Rollup.sol (hunk 2 above)
Comment (krlosMata) [reply contracts/Rollup.sol]: change `abstract contract` to `interface`

**Pull request merged**